packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -231,8 +227,9 @@ function resetComponentStateWhenRemoved(vm: VM) {
         }
         runDisconnectedCallback(vm);
         // Spec: https://dom.spec.whatwg.org/#concept-node-remove (step 14-15)
-        runChildNodesDisconnectedCallback(vm);
-        runLightChildNodesDisconnectedCallback(vm);
+
+        unmountChildren(vm.children, vm.renderRoot);
```

Comment:
It looks like `unmountChildren` calls on `removeVM` which has an [assertion check](https://github.com/salesforce/lwc/blob/95834fd381291d7afc1d1b606994476072ea3469/packages/%40lwc/engine-core/src/framework/vm.ts#L247-L250) and can potentially throw an error.  

I'm not super familiar with this area but it looks like this is a new behavior that might be worth mentioning as a release note or a warning so we don't catch anyone by surprise (unless this assertion is more of an internal check).